doc/acl/policy_based_metering/ACL-Policy-Based-Metering.md

```diff
@@ -0,0 +1,402 @@
+| ------- | ---------- | ---------------------------- | ------------- |
+| 1.0     | 2024-10-13 | Shay Goldshmit (**Marvell**) | Initial Draft |
+
+---
+### Scope
+
+This document describes the design of Policy Base Metering (PBM) feature in SONiC.
+
+#### Out of Scope
+
+- [Policer Counter](https://github.com/sonic-net/SONiC/blob/e3f439dcfe2857540a02e4449fce247d4167b621/doc/policer_counter/PolicerCounter-HLD.md#Architecture-Design) - display and check detailed policer statistics (number of packets marked or dropped based on their color).
+- [Everflow](https://github.com/sonic-net/SONiC/blob/master/doc/everflow/SONiC%20Everflow%20Always-on%20HLD.pdf) - creating and managing policers.
+- [ACL User Defined Table Type Support](https://github.com/sonic-net/SONiC/blob/master/doc/acl/ACL-Table-Type-HLD.md) - ACL feature enhancements with a way user creates with user defined set actions.
+---
+### Definitions/Abbreviations
+
+| Term | Definition                         |
+| ---- | ---------------------------------- |
+| ACL  | Access Control List                |
+| NAT  | Network Address Translation        |
+| SAI  | Switch Abstraction Interface       |
+| CIR  | Committed Information Rate         |
+| CBS  | Committed Burst Size               |
+| PIR  | Peak Information Rate              |
+| PBS  | Peak Burst Size                    |
+
+---
+### Overview
+
+Policer in networking are responsible for **metering** (Monitoring the rate of traffic) and **marking** (Flagging traffic that exceeds defined limits) traffic based on predefined criteria.
+By applying policers to ACL rules, SONiC can effectively control the flow of network traffic, ensuring fairness, optimizing bandwidth utilization, and preventing network congestion.
+
+Usage examples:
+- **Security**: Policy-Based Metering can be used to guard against **network storms or DDoS attacks** by limiting traffic rates.
+- **Quality of Service (QoS)**: It helps in prioritizing critical traffic (like VoIP or video streams) while limiting non-critical traffic (for example: backup services traffic).
+- **Data Center**: ensure fair bandwidth distribution.
+
+---
+### Requirements
+#### Functional Requirements
+- Backward compatibility for existing ACL features - If policer is not set, the system will function as it did previously.
+- Ability to config policers with ACL entries.
+- Support all existing Policer types (Policer mode, meter_type).
+- Support all existing ACL types (ACL table types, ACL stages).
+### Scalability Requirements:
+- Support for multiple ACL entries with associated policers.
+- Query and validate ASIC capabilities dynamically.
+#### CLI Requirements
+- Bind policers with ACL rules.
+- Unbind policer from ACL rules.
+- Show ACLs with policers
+---
+### Architecture Design
+
+No SONiC architecture changes are required as the existing infrastructure is being used.
+
+---
+### High-Level Design
+
+#### Modules and Sub-Modules
+
+- **SWSS**
+  - ACL-Orch
```

Comment:
Need to clearly distinguish between what exists/module role to what is being added